city_scrapers/spiders/il_finance_authority.py

```diff
@@ -0,0 +1,171 @@
+import re
+from datetime import datetime
+from io import BytesIO, StringIO
+
+import scrapy
+from city_scrapers_core.constants import BOARD, COMMISSION, COMMITTEE
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from pdfminer.high_level import extract_text_to_fp
+from pdfminer.layout import LAParams
+from pdfminer.pdfparser import PDFSyntaxError
+
+
+class IlFinanceAuthoritySpider(CityScrapersSpider):
+    name = "il_finance_authority"
+    agency = "Illinois Finance Authority"
+    timezone = "America/Chicago"
+    start_urls = ["https://www.il-fa.com/public-access/board-documents/"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def parse(self, response):
+        for item in response.css("tr:nth-child(1n+2)"):
```

Comment:
We don't have to use this much, but because there are so many meetings here it would be good to use our `CITY_SCRAPERS_ARCHIVE` setting so that we don't have to pull the entire meeting list every time. Here's an example https://github.com/City-Bureau/city-scrapers/blob/c42620480f9184ca74885e2a3c6ef576faba3eb7/city_scrapers/spiders/chi_zoning_board.py#L33, we would probably want to check within 6 months to a year from the current date

Reply:
Can you explain more about the meaning behind this idea? I added it to my code and now it doesnt work. Also more about the settings object? Thank you

Reply:
Sure! We want to be careful about spamming sites with a ton of requests at once, and typically when we scrape a site we're only interested in the last few past meetings and next few upcoming. To reduce the amount of requests we make, as well as simplify the output for anyone using the feeds directly, we try to set ranges of time relative to the current date that we're interested like everything in the past year in that example.

Scrapy's settings are a way for managing configuration across spiders like where the output is written or how quickly requests should be made. You can find more info on them in the [scrapy documentation on settings](https://docs.scrapy.org/en/latest/topics/settings.html).

Could you explain more about what isn't working? It's hard for me to debug without an example, but in general all the `CITY_SCRAPERS_ARCHIVE` settings is doing is giving us a boolean that can be put inside a conditional, so it doesn't have to work the same way as the example